ecommerce/serializers.py

```diff
@@ -286,10 +286,24 @@ def _update_items(cls, basket, items):
             product_id = item.get("id")
             run_ids = item.get("run_ids")
             product = models.Product.objects.get(id=product_id)
+            previous_product = models.Product.objects.filter(
+                basketitem__basket=basket
+            ).first()
+
             if run_ids is not None:
                 runs = cls._get_runs_for_product(
                     product=product, run_ids=run_ids, user=basket.user
                 )
+            elif previous_product is not None and product_id == previous_product.id:
+                # User is updating basket item to the same item as before
+                run_ids = list(
+                    models.CourseRunSelection.objects.filter(basket=basket).values_list(
+                        "run", flat=True
+                    )
+                )
+                runs = cls._get_runs_for_product(
+                    product=product, run_ids=run_ids, user=basket.user
+                )
```

Comment:
Seems like a better way to write this would be:

```python
	        previous_product = models.Product.objects.filter(
                basketitem__basket=basket
            ).first()
            if previous_product is not None and product_id == previous_product.id:
                # User is updating basket item to the same item as before
                run_ids = list(
                    models.CourseRunSelection.objects.filter(basket=basket).values_list(
                        "run", flat=True
                    )
                )

            if run_ids is not None:
                runs = cls._get_runs_for_product(
                    product=product, run_ids=run_ids, user=basket.user
                )
            else:
                runs = None	                runs = None
```

Reply:
Sure, but we will need to check `run_ids is None` in the first if block to make sure the user does not want to update it.